app-header-layout/app-header-layout.html

```diff
@@ -146,36 +146,36 @@
         this.debounce('_resetLayout', this._updateContentPosition);
       },
 
-      _getFixedNodes: function() {
-        return Polymer.dom(this.$.header).getDistributedNodes().
-            concat(Polymer.dom(this.$.underlayContent).getDistributedNodes());
-      },
-
       _updateContentPosition: function() {
         var header = this.header;
-        var containerStyle = this.$.contentContainer.style;
-
         if (!this.isAttached || !header) {
           return;
         }
 
-        var rect = this.getBoundingClientRect();
-        var rightOffset = document.documentElement.clientWidth - rect.right;
-        this._headerHeight = header.offsetHeight;
-
+        // Update the content container position.
+        var headerHeight = header.offsetHeight;
+        var containerStyle = this.$.contentContainer.style;
         if (header.fixed && !header.willCondense() && this.hasScrollingRegion) {
-          containerStyle.marginTop = this._headerHeight + 'px';
+          // If the header size does not change and we're using a scrolling region, exclude
+          // the header area from the scrolling region so that the header doesn't overlap
+          // the scrollbar.
+          containerStyle.marginTop = headerHeight + 'px';
           containerStyle.paddingTop = '';
         } else {
-          containerStyle.paddingTop = this._headerHeight + 'px';
+          containerStyle.paddingTop = headerHeight + 'px';
           containerStyle.marginTop = '';
         }
 
-        // update fixed nodes
-        this._getFixedNodes().forEach(function(node) {
-          node.style.left = this.hasScrollingRegion ? '' : rect.left + 'px';
-          node.style.right = this.hasScrollingRegion ? '' : rightOffset + 'px';
-        }.bind(this));
+        // Update the header position.
+        if (!this.hasScrollingRegion) {
+          var rect = this.getBoundingClientRect();
```

Comment:
We should move this line to the top so we can batch all the reads together.

See https://github.com/PolymerLabs/app-layout/pull/137

```
var headerHeight = header.offsetHeight; // read
var rect = this.getBoundingClientRect(); // read
containerStyle.marginTop = this._headerHeight + 'px';  // write
header.style.left = rect.left + 'px';  // write
```